day_two/program.go

```diff
@@ -44,115 +60,38 @@ const (
 )
 
 func main() {
-	parsedInput := parseInput(input)
+	parsedInput, err := ParseFile(os.Args[1])
+	if err != nil {
+		log.Fatalf("parse input: %v", err)
+	}
 
-	partOneOutput := partOne(parsedInput)
-	partTwoOutput := partTwo(parsedInput)
+	partOneOutput := PartOne(parsedInput)
+	partTwoOutput := PartTwo(parsedInput)
 	fmt.Println(partOneOutput)
 	fmt.Println(partTwoOutput)
 }
 
-func partTwo(input []Play) (totalScore int) {
-	choices := map[string]string{
-		"A": "Rock",
-		"B": "Paper",
-		"C": "Scissors",
-		"X": "Loss",
-		"Y": "Draw",
-		"Z": "Win",
-	}
-
-	scores := map[string]int{
-		"Rock":     Rock,
-		"Paper":    Paper,
-		"Scissors": Scissors,
-	}
+func PartTwo(input []Play) (score int) {
+	results := [...]int{Loss, Draw, Win}
```

Comment:
While this is a cool way of doing this, personally I wouldn't ship this to production. 
The way it was set up initially left the code in a way that someone with no idea who had to fix a bug would be able to find and fix the issue in moments. 

On that note.. i assume this line is just unpacking the results somehow? Not sure what `[...]` is doing here?

Reply:
I agree, it's not the clearest, though with some comments it would be better. I rewrote it in part because I found the original `switch`-based implementation with maps to be confusing, especially because one of the maps, `scores`, didn't actually do anything in any case other than `Draw`.

The `[...]` syntax creates an automatically sized array. In Go, arrays and slices are different things, and arrays of different sizes are also different things. In other words, `[]int`, `[2]int`, and `[3]int` are all completely different, incompatible types. You can define a slice literal using `[]int{1, 2, 3}` and an array literal using `[3]int{1, 2, 3}`, but it can be annoying to have to manually size the array literal, so replacing the number with `...` results in an array with a size automatically determined by the number of elements given.